### ISSUE TITLE
.gitignore misses out some items

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6736]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#6736|
|Type|enhancement|
|Created|2013-09-09 18:10:15Z|

Building off [#6731](https://github.com/twisted/twisted/issues/6731), there are a few things missing:

 - trial makes sequenced directories in case two are running in the directory at the same time
 - there is no dist/, in case anyone makes a source dist (I am not sure of the applicability of this to Twisted, but I think it might be nice to have anyway)
 - There are some other things in http://twistedmatrix.com/trac/wiki/GitMirror that must be in there for a reason.

Since .gitignore just ignores things, I think if we include everything here, it won't hurt.

Attachments:

* [gitignore-additions-6736.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/24a/24afbd046f7aad56041a5ba2fd2f289ad3156487/e05f9d2cfd0a96c60e99ea9eef6feab01836db0a.diff) (254 bytes) - added by hawkowl on 2013-09-09 18:18:21Z - Patch that adds extra items to the .gitignore file
* [gitignore-additions-6736-v2.diff](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/24a/24afbd046f7aad56041a5ba2fd2f289ad3156487/954f3441a238660dbfcf468732d4b3a3feda4854.diff) (221 bytes) - added by hawkowl on 2013-09-09 19:30:28Z - Patch that adds extra items to the .gitignore file - v2

<details><summary>Searchable metadata</summary>

```
trac-id__6736 6736
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1378750215000000 1378750215000000
changetime__1378755994000000 1378755994000000
version__None None
owner__exarkun exarkun
cc__hawkowl@...
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl removed owner|
|-|-|

Patch uploaded with the changes as mentioned in the description. Based off current trunk.

I also ordered it alphabetically so it looks nice.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun set owner to @hawkowl|
|-|-|

Hi hawkowl.  Thanks for these refinements.  However...

  1. There are no `.la` or `.lo` or `.a` build artifacts related to builds of Twisted.
  1. Neither is there ever a top-level `dist` directory.
  1. The only "*.cache" file that is ever produced is now "dropin.cache" which also now seems to be explicitly included.  So I don't know why "*.cache" needs to be there.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

Also, with a gitignore in the repository, the directions on the wiki page you linked to are now superfluous.  Please remove them.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl removed owner|
|-|-|

Thanks exarkun for the review and for fixing up the description.

New patch uploaded - this one is off trunk, again, with the review suggestions.

I'll also get right on the wiki changes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @exarkun|
|-|-|

Okay.  I can see the motivation for each of these additions and removals.  I'll apply the patch to trunk.  Thank you.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [39920]) Apply gitignore-additions-6736-v2.diff

Author: hawkowl
Reviewer: exarkun
Fixes: [#6736](https://github.com/twisted/twisted/issues/6736)

Make some miscellaneous changes to the .gitignore file.
